### PR TITLE
Geometry checker improvements

### LINKED ISSUE #44247: Geometry Checker should remember all its settings
**Feature description.**
While most settings are stored and recalled on the Geometry Checker core plugin after a restart, there are a few that are not:
- *Tolerance* value
- *Modify input layers* vs *Create new layers*
- *Format*
- *Output directory*
- *Filename prefix*

### LINKED ISSUE #44251: Geometry Checker should also export the Value field
**Feature description.**
The check geometries core plugin has a *Value* field on its results containing a value related to some errors, like gap / overlap area, minimal angle, segment length etc.
It would be very useful if that value was also exported when exporting the results to a file.

### PR DESCRIPTION
**Pull request by uclaros**
## Description
This PR implements the following improvements to the geometry checker ux:
- Exports the *Value* field when exporting errors (ie gap/overlap area)
- ~Highlights and exports a *pointOnSurface* instead of the *centroid* for gaps/overlaps/areas so that zooming has the error visible.~
- Store and recall the *export path*, the *tolerance* and the *modify input layers* radio.

Fixes #44247
Fixes #44251

### CODE REVIEW

File: src/plugins/geometry_checker/qgsgeometrycheckerresulttab.cpp [hunk 1]
```diff
@@ -222,8 +222,10 @@ void QgsGeometryCheckerResultTab::updateError( QgsGeometryCheckError *error, boo
 
 void QgsGeometryCheckerResultTab::exportErrors()
 {
-  QString initialdir;
-  QDir dir = QFileInfo( mChecker->featurePools().first()->layer()->dataProvider()->dataSourceUri() ).dir();
+  QString initialdir = QgsSettings().value( "/geometry_checker/previous_values/exportDirectory", "" ).toString();
```
Comment (nyalldawson) [inline src/plugins/geometry_checker/qgsgeometrycheckerresulttab.cpp:225]: This should fallback to QDir::homePath() instead of ""

**Review by nyalldawson — APPROVED** [review]

**Review by lbartoletti — APPROVED** [review]

**Comment by uclaros** [comment]
Before adapting the tests I'd like to make sure that *pointOnSurface* is indeed deterministic!

**Comment by elpaso** [comment]
@uclaros is this good for merge or are you still working on it?

**Comment by uclaros** [comment]
I had some test cases failing on my machine that I assumed were caused by my local geos with OvelayNG. Now the tests pass on CI so my assumption was probably correct, but I would like to give it a last look, since the error location coordinates are actually also used in error resolution and not only for display purposes.

**Comment by uclaros** [comment]
Ok, I reverted the *centroid* vs *pointOnSurface* part.
I'll tackle that on a different PR.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**